iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -156,11 +163,19 @@ public void abortTask(TaskAttemptContext originalContext) throws IOException {
     TaskAttemptContext context = TezUtil.enrichContextWithAttemptWrapper(originalContext);
 
     // Clean up writer data from the local store
-    Map<String, HiveIcebergRecordWriter> writers = HiveIcebergRecordWriter.removeWriters(context.getTaskAttemptID());
+    Map<String, HiveIcebergWriter> writers = HiveIcebergWriter.getRecordWriters(context.getTaskAttemptID());
```

Comment:
Maybe the place of this should not be here, but in the respecitve writers, or in the Committer?

Reply:
We could move the logic to a static method in the writer, e.g. `HiveIcebergWriter.closeAndRemoveWriters(attemptId)`. Is this what you had in mind? This way we could call this static method in both `commitTask` and `abortTask`.

Actually I don't see the writer closing logic in the `commitTask` method currently (only in `abortTask`), so it's possible we're not even closing out our iceberg writers currently in the happy path?

Reply:
As discussed, let's use a single writer map for both DeleteWriters and RecordWriters